.github/workflows/ci.yml

```diff
@@ -170,9 +190,17 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        multiverse: [agent, api, background, background_2, database, httpclients, rails, rest, serialization, sinatra]
-        ruby-version: [2.0.0-p648, 2.1.10, 2.2.10, 2.3.8, 2.4.10, 2.5.8, 2.6.6, 2.7.1, jruby-9.2.12.0]
+        multiverse: [agent, api, background, background_2, database, httpclients, rails, rails_extras, rest, serialization, sinatra]
+        ruby-version: [2.0.0-p648, 2.1.10, 2.2.10, 2.3.8, 2.4.10, 2.5.8, 2.6.6, 2.7.1, 3.0.0-preview1, jruby-9.2.12.0]
         exclude:
+          - ruby-version: "3.0.0-preview1"
+            multiverse: api
+          - ruby-version: "3.0.0-preview1"
+            multiverse: sinatra
+          - ruby-version: "3.0.0-preview1"
+            multiverse: rails
+          - ruby-version: "3.0.0-preview1"
+            multiverse: rails_extras
```

Comment:
For api and sinatra, it is excluded on 2.7 as well, since there are some gems don't work on 2.7 either, so it wouldn't work on 3.0. 
For rails and rails_extras, there aren't any rails that work with ruby 3.0, so these will need to be excluded until that releases.